src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as propTypes from 'prop-types';
+import omit = require('lodash/omit');
+
+import FormFieldSpinnerUp from 'wix-ui-icons-common/system/FormFieldSpinnerUp';
+import FormFieldSpinnerDown from 'wix-ui-icons-common/system/FormFieldSpinnerDown';
+
+import {
+  TimePicker as CoreTimePicker,
+  TimePickerProps as CoreTimePickerProps,
+} from 'wix-ui-core/TimePicker';
+import {AmPmOptions} from 'wix-ui-core/dist/src/components/TimePicker/constants';
+import style from './TimePicker.st.css';
+import {Size} from './constants';
+
+export interface TimePickerPropsExtended {
+  size?: Size,
+  value?: String;
+  disableAmPm?: boolean;
+  width?: string;
+}
+
+/*
+  'CoreTimePickerProps.useAmPm' prop must not be included in TimePickerProps.
+  Use type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>> to omit it from CoreTimePickerProps.
+  TODO: after typescript is updated to ^2.9 omit 'useAmPm' prop
+ */
+export type TimePickerProps = CoreTimePickerProps & TimePickerPropsExtended;
```

Comment:
Did you try to use the typescript `Omit` function?
https://github.com/wix/wix-ui-backoffice/blob/34004fc73b51877d5ca9caaf968f1b27ff6269b3/src/components/CircularProgressBar/CircularProgressBar.tsx#L18